File: example-nonblocking/src/main/java/com/davin111/examplenonblocking/ContentRepository.java
package com.davin111.examplenonblocking;

import org.springframework.data.repository.reactive.ReactiveCrudRepository;
import org.springframework.stereotype.Repository;
import reactor.core.publisher.Flux;

@Repository
public interface ContentRepository extends ReactiveCrudRepository<Content, Long> {

    Flux<Content> findByIsPremiumFalse();

    Flux<Content> findByIsPremiumTrue();
}

File: example-nonblocking-kotlin/src/main/kotlin/com/davin111/examplenonblockingkotlin/ContentRepository.kt
package com.davin111.examplenonblockingkotlin

import org.springframework.data.repository.kotlin.CoroutineCrudRepository
import org.springframework.stereotype.Repository

@Repository
interface ContentRepository : CoroutineCrudRepository<Content, Long> {
    suspend fun findByIsPremiumFalse(): List<Content>
    suspend fun findByIsPremiumTrue(): List<Content>
}
